README.md

```diff
@@ -275,6 +275,57 @@ class RecentPostReference < ActiveRecord::Base
 end
 ```
 
+## Custom Preloads
+
+In addition to preloading relations, you can also define custom preloads directly in your model. The only requirement is that you need to be able perform a lookup for multiple records/ids and return a single Hash with the ids as keys.
+If that's the case, these preloads can nearly be anything. Some examples could be:
+
+- simple aggregations (count, sum, maximum, etc.)
+- more complex custom SQL queries
+- external API requests (ElasticSearch, Redis, etc.)
+- relations with primary_keys stored in a `jsonb` column
+
+Here's how:
+
+```ruby
+class Blog < ActiveRecord::Base
+  has_many :posts
+  
+  def posts_count
+    goldiload(:posts_count) do |ids|
```

Comment:
That's a nice idea! What do you think about using sth like `block.source_location.join(':')` (maybe also hashing it) as the cache name?

Reply:
I've implemented it via `block.source_location.join(':')` but kept the `cache_name` as an optional argument. I can hardly come up with a case where the source location would cause any issues (heavy meta programming proc factories??), but keeping the optional argument also doesn't add any dept or complexity imho.